test/FunctionsV2/OutOfProcTests.cs

```diff
@@ -122,5 +138,209 @@ public async Task CallHttpActionOrchestrationWithManagedIdentity()
+                localRpcEndpointEnabled: localRcpEnabled,
+                notificationUrl: testNotificationUrl))
+            {
+                await host.StartAsync();
+
+                // Fetch the JSON that was passed to the DurableClient binding
+                string[] jsonRef = new string[1];
+                await host.CallAsync(
+                    nameof(ClientFunctions.GetDurableClientConfigJson),
+                    new Dictionary<string, object>
+                    {
+                        { "jsonRef", jsonRef },
+                    });
+
+                string jsonString = jsonRef[0];
+                Assert.NotNull(jsonString);
+                JObject outerJson = JObject.Parse(jsonString);
+
+                // Expected format:
+                // {
+                //   "taskHubName": "BindToDurableClientAsStringV2",
+                //   "creationUrls": {
+                //     "createNewInstancePostUri": "https://durable.edu/runtime/webhooks/durabletask/orchestrators/{functionName}[/{instanceId}]",
+                //     "createAndWaitOnNewInstancePostUri": "https://durable.edu/runtime/webhooks/durabletask/orchestrators/{functionName}[/{instanceId}]?timeout={timeoutInSeconds}&pollingInterval={intervalInSeconds}"
+                //   },
+                //   "managementUrls": {
+                //     "id": "INSTANCEID",
+                //     "statusQueryGetUri": "https://durable.edu/runtime/webhooks/durabletask/instances/INSTANCEID?taskHub=BindToDurableClientAsStringV2&connection=Storage",
+                //     "sendEventPostUri": "https://durable.edu/runtime/webhooks/durabletask/instances/INSTANCEID/raiseEvent/{eventName}?taskHub=BindToDurableClientAsStringV2&connection=Storage",
+                //     "terminatePostUri": "https://durable.edu/runtime/webhooks/durabletask/instances/INSTANCEID/terminate?reason={text}&taskHub=BindToDurableClientAsStringV2&connection=Storage",
+                //     "rewindPostUri": "https://durable.edu/runtime/webhooks/durabletask/instances/INSTANCEID/rewind?reason={text}&taskHub=BindToDurableClientAsStringV2&connection=Storage",
+                //     "purgeHistoryDeleteUri": "https://durable.edu/runtime/webhooks/durabletask/instances/INSTANCEID?taskHub=BindToDurableClientAsStringV2&connection=Storage"
+                //   },
+                //   "baseUrl": "https://durable.edu/runtime/webhooks/durabletask/",
+                //   "requiredQueryStringParameters": "code=abcdefg",
+                //   "rpcBaseUrl": "http://127.0.0.1:17071/durabletask/" (or null)
+                // }
+
+                Assert.True(outerJson.TryGetValue("taskHubName", out JToken taskHubName));
+                Assert.StartsWith(nameof(this.BindToDurableClientAsString), (string)taskHubName);
+
+                Func<JToken, string, string[], Uri> commonUriValidation = (
+                    JToken json,
+                    string fieldName,
+                    string[] requiredSegments) =>
```

Comment:
Not quite sure what we gain from this requiredSegments approach instead of just constructing the expected URI. The URIs seem deterministic, so it seems like we should use those instead of just checking if the path contains some segments.

Reply:
Might have been an artifact from a previous iteration. I'll see if I can clean it up.

Reply:
I simplified the structure of this test quite a bit, but constructing the full URLs were going to be more work than I'm willing to do at this point. I'm okay with revisiting this next time we need to update this code if we think stricter validation will improve the quality of this test.